tests/scripts/thread-cert/thread_cert.py

```diff
@@ -153,7 +157,11 @@ def _setUp(self):
             params = self._parse_params(params)
             initial_topology[i] = params
 
+            # backbone_network_name is None if no backbone network is defined
+            backbone_network_name = self._determine_backbone_network_name(params.get('backbone_network'))
```

Comment:
does the following work? You may wrap it as a method.
```suggestion
            backbone_network_name = f'backbone{PORT_OFFSET}.{backbone_network}'
```

Reply:
The function `_determine_backbone_network_name()` and other logic inside are needed, as the return value of `_determine_backbone_network_name()` has three cases:
1. returns `None` when the node has no backbone traffic (otbr node or host node);
2. returns string with `backbone{PORT_OFFSET}` format when the test case has only one backbone network defined;
3. returns string with `backbone{PORT_OFFSET}.{backbone_num}` format when the test case has multiple backbone network defined.

Also `_determine_backbone_network_name()` has multiple error checks which help return early and give clear reason if there's any internal error (e.g. bad TOPOLOGY definition).

The idea to return `f'backbone{PORT_OFFSET}.{backbone_network}'` in case#3 is good, I will update.

Reply:
updated here https://github.com/openthread/openthread/pull/10596/commits/a1f761ba43657d3866362ca433662820a1f4160b#diff-e25877ffbdc91edb2cda7e4736b5abff5b12d9cfe5b13dc28c943d752b5a6e94R503

Reply:
Is it necessary to differentiate these case 2 and case 3? It looks like unifying the format `f'backbone{PORT_OFFSET}.{backbone_network}'` simplifies the code.

Reply:
+1 on @bukepo's suggestion to unify case #2 and case #3. Maybe you can use 0 as the default parameter of `backbone_network_id`.

Reply:
Also, I think case #1 only happens when the node doesn't need backbone network. Probably you can skip this step when the node type is not OtbrNode or HostNode.

Reply:
Thanks, updated per comment. merged case#2 and case#3 into single format. 
see https://github.com/openthread/openthread/pull/10596/files#diff-e25877ffbdc91edb2cda7e4736b5abff5b12d9cfe5b13dc28c943d752b5a6e94R160